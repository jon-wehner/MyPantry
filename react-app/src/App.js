import React, { useState, useEffect } from "react";
import {Route, Switch } from "react-router-dom";
import { useDispatch } from "react-redux"
import LoginForm from "./components/auth/LoginForm";
import SignUpForm from "./components/auth/SignUpForm";
import NavBar from "./components/NavBar/";
import ProtectedRoute from "./components/auth/ProtectedRoute";
import ShoppingList from "./components/ShoppingList"
import { authenticate } from "./store/session";
import LandingPage from "./LandingPage";
import Dashboard from "./components/Dashboard";
import Inventory from "./components/Inventory";

function App() {
  const dispatch = useDispatch()
  const [authenticated, setAuthenticated] = useState(false);
  const [loaded, setLoaded] = useState(false);

  useEffect(() => {
    (async() => {
      const user = await dispatch(authenticate());
      if (!user.errors) {
        setAuthenticated(true);
      }
      setLoaded(true);
    })();
  }, [dispatch]);

  if (!loaded) {
    return null;
  }

  return (
    <>
      <NavBar authenticated={authenticated} setAuthenticated={setAuthenticated} />
      <Switch>
        <Route path="/login" exact={true}>
          <LoginForm
            authenticated={authenticated}
            setAuthenticated={setAuthenticated}
          />
        </Route>
        <Route path="/sign-up" exact={true}>
          <SignUpForm authenticated={authenticated} setAuthenticated={setAuthenticated} />
        </Route>
        <ProtectedRoute path="/shopping-lists/:id" exact={true} authenticated={authenticated}>
          <ShoppingList />
        </ProtectedRoute>
        <Route path="/" exact={true} authenticated={authenticated}>
          {authenticated ? <Dashboard authenticated={authenticated}/> : <LandingPage authenticated={authenticated} setAuthenticated={setAuthenticated}/>}
        </Route>
        <Route path="/user/:id/inventory">
          <Inventory />
        </Route>
      </Switch>
    </>
  );
}

export default App;
