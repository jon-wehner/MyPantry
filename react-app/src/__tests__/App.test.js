import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import App from '../App';

afterEach(cleanup);

it('should take a snapshot and render the app component', () => {
  const initialState = {};
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const store = mockStore(initialState);
  const { asFragment } = render(<Provider store={store}><App /></Provider>);

  expect(asFragment(<App />)).toMatchSnapshot();
});
