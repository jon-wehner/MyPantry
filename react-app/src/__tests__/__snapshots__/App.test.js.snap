// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot and render the app component 1`] = `<DocumentFragment />`;
